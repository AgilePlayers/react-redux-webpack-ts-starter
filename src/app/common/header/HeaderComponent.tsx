import React from 'react';
import NavbarComponent from './components/navbar/NavbarComponent';
import './header.scss';
import logo from './_assets/img/react.svg';

const HeaderComponent: React.SFC = () => (
  <div className="header-wrapper">
    <div className="header">
      <img src={logo} className="header-logo" alt="logo" />
      <NavbarComponent />
    </div>
  </div>
);

export default HeaderComponent;
