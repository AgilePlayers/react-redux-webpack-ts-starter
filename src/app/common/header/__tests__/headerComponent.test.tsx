import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import HeaderComponent from '../HeaderComponent';

describe('common / header / HeaderComponent', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<HeaderComponent />);
  });

  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contains elements with "header" and "header-logo" class', () => {
    expect(component.find('div.header').length).toEqual(1);
    expect(component.find('img.header-logo').length).toEqual(1);
  });
});
