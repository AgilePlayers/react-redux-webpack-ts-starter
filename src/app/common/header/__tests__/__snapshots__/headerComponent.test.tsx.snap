// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common / header / HeaderComponent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header"
      >
        <img
          alt="logo"
          className="header-logo"
          src="file-stub"
        />
        <NavbarComponent />
      </div>,
      "className": "header-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="header-logo"
            src="file-stub"
          />,
          <NavbarComponent />,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "header-logo",
            "src": "file-stub",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header"
        >
          <img
            alt="logo"
            className="header-logo"
            src="file-stub"
          />
          <NavbarComponent />
        </div>,
        "className": "header-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="header-logo"
              src="file-stub"
            />,
            <NavbarComponent />,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "header-logo",
              "src": "file-stub",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
